src/test/java/unit/Group.java

```diff
@@ -0,0 +1,318 @@
+        "--policy=MEMBER");
+
+    // `terra groups delete --name=$name` as admin
+    groupCreator.login();
+    TestCommand.runCommandExpectSuccess("groups", "delete", "--name=" + name);
+  }
+
+  @Test
+  @DisplayName("list-users reflects adding and removing a user")
+  void listUsersReflectsAddRemove() throws IOException {
+    TestUsers groupCreator = TestUsers.chooseTestUser();
+    groupCreator.login();
+
+    // `terra groups create --name=$name`
+    String name = SamGroups.randomGroupName();
+    TestCommand.runCommandExpectSuccess("groups", "create", "--name=" + name);
+
+    // track the group so we can clean it up in case this test method fails
+    trackedGroups.trackGroup(name, groupCreator);
+
+    // `terra groups add-user --name=$name --email=$email --policy=MEMBER`
+    TestUsers groupMember = TestUsers.chooseTestUserWhoIsNot(groupCreator);
+    TestCommand.runCommandExpectSuccess(
+        "groups", "add-user", "--name=" + name, "--email=" + groupMember.email, "--policy=MEMBER");
+
+    // check that group member is included in the list-users output with one policy
+    expectListedMemberWithPolicies(name, groupMember.email, GroupPolicy.MEMBER);
+
+    // `terra groups add-user --name=$name --email=$email --policy=ADMIN`
+    TestCommand.runCommandExpectSuccess(
+        "groups", "add-user", "--name=" + name, "--email=" + groupMember.email, "--policy=ADMIN");
+
+    // check that group member is included in the list-users output with two policies
+    expectListedMemberWithPolicies(name, groupMember.email, GroupPolicy.MEMBER, GroupPolicy.ADMIN);
+
+    // `terra groups remove-user --name=$name --email=$email --policy=MEMBER`
+    TestCommand.runCommandExpectSuccess(
+        "groups",
+        "remove-user",
+        "--name=" + name,
+        "--email=" + groupMember.email,
+        "--policy=MEMBER");
+
+    // check that group member is included in the list-users output with one policy
+    expectListedMemberWithPolicies(name, groupMember.email, GroupPolicy.ADMIN);
+
+    // `terra groups remove-user --name=$name --email=$email --policy=ADMIN`
+    TestCommand.runCommandExpectSuccess(
+        "groups",
+        "remove-user",
+        "--name=" + name,
+        "--email=" + groupMember.email,
+        "--policy=ADMIN");
+
+    // check that group member is no longer included in the list-users output
+    Optional<UFGroupMember> listedGroupMember = listMembersWithEmail(name, groupMember.email);
+    assertTrue(listedGroupMember.isEmpty(), "test user is no longer included in members list");
+
+    // check that the group creator is included in the list-users output
+    expectListedMemberWithPolicies(name, groupCreator.email, GroupPolicy.ADMIN);
+
+    // `terra groups delete --name=$name`
+    TestCommand.runCommandExpectSuccess("groups", "delete", "--name=" + name);
+  }
+
+  @Test
+  @DisplayName("cli-testers group includes the appropriate test users")
+  void cliTestersGroupMembership() throws IOException {
```

Comment:
maybe worth a comment that this isn't about testing the CLI so much as testing the environment setup?

Reply:
added